pkg/cloudprovider/provider/hetzner/provider.go

```diff
@@ -183,6 +192,14 @@ func (p *provider) Validate(spec v1alpha1.MachineSpec) error {
 		}
 	}
 
+	if len(c.Firewalls) != 0 {
+		for _, firewall := range c.Firewalls {
+			if _, _, err = client.Firewall.Get(ctx, firewall); err != nil {
+				return fmt.Errorf("failed to get firewall %q: %v", firewall, err)
+			}
```

Comment:
```suggestion
			if frwl, _, err := client.Firewall.Get(ctx, firewall); err != nil || frwl == nil {
				return fmt.Errorf("failed to get firewall %q: %v", firewall, err)
			}
```

Reply:
this is how hcloud handles the 404: https://github.com/hetznercloud/cli/blob/bb30002002cd2c8af6b20269eff549d09f7204a5/internal/cmd/firewall/describe.go#L36-L42

404 of object doesn’t return an error but rather a nil object and we have to check for nilness

Reply:
I'll change now. To pass spell checks, do you mind if I use fetchedFirewall instead of frwl?

Reply:
sure, I used frwl just as a demo that there has to be something

Reply:
OK cool.